flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NotPatternITCase.java

```diff
@@ -980,6 +980,53 @@ public void testNotFollowedByBeforeZeroOrMoreCombinationsSkipTillAny() throws Ex
 		));
 	}
 
+	@Test
+	public void testNotFollowedByInTheEndOfGroupPattern() throws Exception {
+
+		List<StreamRecord<Event>> inputEvents = new ArrayList<>();
+
+		int i = 0;
+		inputEvents.add(new StreamRecord<>(NotFollowByData.A_1, i++));
+		inputEvents.add(new StreamRecord<>(NotFollowByData.C_1, i++));
+		inputEvents.add(new StreamRecord<>(NotFollowByData.B_1, i++));
+		inputEvents.add(new StreamRecord<>(NotFollowByData.A_1, i++));
+		inputEvents.add(new StreamRecord<>(NotFollowByData.B_2, i++));
+
+		Pattern<Event, ?> startPattern = Pattern
+				.<Event>begin("a").where(new SimpleCondition<Event>() {
+					private static final long serialVersionUID = 5726188262756267490L;
+
+					@Override
+					public boolean filter(Event value) throws Exception {
+						return value.getName().equals("a");
+					}
+				})
+				.notFollowedBy("not c").where(new SimpleCondition<Event>() {
+					private static final long serialVersionUID = 5726188262756267490L;
```

Comment:
For tests, I think it is nicer to have a class-wise `@SuppressWarning("serial")` rather than having these IDs in every function.

Reply:
OK